shared/utility/motion/splines/TrajectoryUtils.hpp

```diff
@@ -87,52 +88,52 @@ namespace utility::motion::splines {
         }
 
         // Operators
-        bool operator<(TrajectoryTypes const& other) const {
+        [[nodiscard]] constexpr bool operator<(TrajectoryTypes const& other) const {
             return value < other.value;
         }
-        bool operator>(TrajectoryTypes const& other) const {
+        [[nodiscard]] constexpr bool operator>(TrajectoryTypes const& other) const {
             return value > other.value;
         }
-        bool operator<=(TrajectoryTypes const& other) const {
+        [[nodiscard]] constexpr bool operator<=(TrajectoryTypes const& other) const {
             return value <= other.value;
         }
-        bool operator>=(TrajectoryTypes const& other) const {
+        [[nodiscard]] constexpr bool operator>=(TrajectoryTypes const& other) const {
             return value >= other.value;
         }
-        bool operator==(TrajectoryTypes const& other) const {
+        [[nodiscard]] constexpr bool operator==(TrajectoryTypes const& other) const {
             return value == other.value;
         }
-        bool operator!=(TrajectoryTypes const& other) const {
+        [[nodiscard]] constexpr bool operator!=(TrajectoryTypes const& other) const {
             return value != other.value;
         }
-        bool operator<(TrajectoryTypes::Value const& other) const {
+        [[nodiscard]] constexpr bool operator<(TrajectoryTypes::Value const& other) const {
             return value < other;
         }
-        bool operator>(TrajectoryTypes::Value const& other) const {
+        [[nodiscard]] constexpr bool operator>(TrajectoryTypes::Value const& other) const {
             return value > other;
         }
-        bool operator<=(TrajectoryTypes::Value const& other) const {
+        [[nodiscard]] constexpr bool operator<=(TrajectoryTypes::Value const& other) const {
             return value <= other;
         }
-        bool operator>=(TrajectoryTypes::Value const& other) const {
+        [[nodiscard]] constexpr bool operator>=(TrajectoryTypes::Value const& other) const {
             return value >= other;
         }
-        bool operator==(TrajectoryTypes::Value const& other) const {
+        [[nodiscard]] constexpr bool operator==(TrajectoryTypes::Value const& other) const {
             return value == other;
         }
-        bool operator!=(TrajectoryTypes::Value const& other) const {
+        [[nodiscard]] constexpr bool operator!=(TrajectoryTypes::Value const& other) const {
             return value != other;
         }
 
         // Conversions
-        operator Value() const {
+        [[nodiscard]] constexpr operator Value() const {
             return value;
         }
-        operator uint8_t() const {
+        [[nodiscard]] constexpr operator uint8_t() const {
             return value;
         }
 
-        operator std::string() const {
+        [[nodiscard]] operator std::string() const {
```

Comment:
Definition in `.cpp` please

Reply:
why?

Reply:
ODR

Reply:
Granted, you have showed me why that's true for the general case, but this case is for a "smart enum" struct which doesn't even have its own hpp. I suppose I'll make it inline?